py/server/tests/test_table.py

```diff
@@ -489,30 +489,32 @@ def test_partitioned_agg_by(self):
             self.assertEqual(result_pt2.table.size, 2)
             self.assertEqual(result_pt.keys().to_string(), result_pt2.keys().reverse().to_string())
 
-    def test_snapshot(self):
-        with self.subTest("do_init is False"):
-            t = empty_table(0).update(
-                formulas=["Timestamp=io.deephaven.time.DateTime.now()", "X = i * i", "Y = i + i"]
-            )
-            snapshot = t.snapshot(source_table=self.test_table)
+    def test_snapshot_when(self):
+        t = empty_table(0).update(
+            formulas=["Timestamp=io.deephaven.time.DateTime.now()", "X = i * i", "Y = i + i"]
+        )
+        with self.subTest("with defaults"):
+            snapshot = self.test_table.snapshot_when(t)
             self.assertEqual(len(t.columns) + len(self.test_table.columns), len(snapshot.columns))
             self.assertEqual(0, snapshot.size)
 
-        with self.subTest("do_init is True"):
-            snapshot = t.snapshot(source_table=self.test_table, do_init=True)
+        with self.subTest("initial=True"):
+            snapshot = self.test_table.snapshot_when(t, initial=True)
             self.assertEqual(self.test_table.size, snapshot.size)
 
-        with self.subTest("with cols"):
-            snapshot = t.snapshot(source_table=self.test_table, cols="X")
+        with self.subTest("cols=\"X\""):
+            snapshot = self.test_table.snapshot_when(t, stamp_cols="X")
             self.assertEqual(len(snapshot.columns), len(self.test_table.columns) + 1)
-            snapshot = t.snapshot(source_table=self.test_table, cols=["X", "Y"])
+
+        with self.subTest("cols=[\"X\", \"Y\"]"):
+            snapshot = self.test_table.snapshot_when(t, stamp_cols=["X", "Y"])
```

Comment:
What is the behavior if `stamp_cols` is not set?  Everything?  Nothing?  Ideally, we want a way to do everything and a way to do nothing.

Reply:
Test case for including no stamp cols?

Reply:
I've added docs - when stamp_cols is None or empty, all colunms are included. This case is tested as part of the "with defaults" subtest.